homeassistant/components/kermi/water_heater.py

```diff
@@ -0,0 +1,169 @@
+"""Kermi Water Heater accessed via an IFM modbus gateway."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.components import water_heater
+from homeassistant.components.water_heater import (
+    WaterHeaterEntity,
+    WaterHeaterEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_WHOLE, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import MODBUS_REGISTERS
+
+SUPPORT_FLAGS_HEATER = (
+    WaterHeaterEntityFeature.TARGET_TEMPERATURE
+    | WaterHeaterEntityFeature.OPERATION_MODE
+)
+
+DOMAIN = "kermi"
+logger = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the config entry."""
+    kermi_water_heater = KermiWaterHeater(
+        "Kermi x-buffer",
+        entry,
+        hass.data[DOMAIN][entry.entry_id]["coordinator"],
+    )
+    async_add_entities([kermi_water_heater])
+
+
+class KermiWaterHeater(WaterHeaterEntity, CoordinatorEntity):
+    """Representation of water_heater device attached to IFM."""
+
+    _attr_should_poll = False
```

Comment:
```suggestion
    _attr_should_poll = False
    _attr_has_entity_name = True
```
is required for new integrations